HeterogeneousCore/CUDACore/src/chooseCUDADevice.cc

```diff
@@ -0,0 +1,18 @@
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "HeterogeneousCore/CUDAServices/interface/CUDAService.h"
+
+#include "chooseCUDADevice.h"
+
+namespace cudacore {
+  int chooseCUDADevice(edm::StreamID id) {
+    edm::Service<CUDAService> cudaService;
+
+    // For startes we "statically" assign the device based on
+    // edm::Stream number. This is suboptimal if the number of
+    // edm::Streams is not a multiple of the number of CUDA devices
+    // (and even then there is no load balancing).
+    //
+    // TODO: improve the "assignment" logic
+    return id % cudaService->numberOfDevices();
```

Comment:
what is (will be) the interplay with the `CUDAService`'s `numberOfStreamsPerDevice` ?

Reply:
Thanks for reminding. `numberOfStreamsPerDevice` would be removed as there is no way to support that with a "beginning of job" decision for whether to use GPU or not. If such a functionality would really be needed, it would be one argument (more) for implementing event-by-event decisions for SwitchProducer (and to be implemented later).

Reply:
`numberOfStreamsPerDevice` functionality is removed in #282.